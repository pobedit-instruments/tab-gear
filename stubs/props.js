import { bindActionCreators } from 'redux';

import actual from './tabs';
import * as Actions from '../views/actions';
import * as ActionTypes from '../views/constants/ActionTypes';
import config from '../config';

export const tabs = {
	actual,
	masked: false,
	active: actual.find(tab => tab.active),
	search: actual
};

export const type = ActionTypes.SHOW_TABS;

export const actions = new function () {
	let actions = {};

	for (let [name, value] of Object.entries(Actions)) {
		actions[name] = bindActionCreators(value, () => {});
	}

	return actions;
};

export { config };
