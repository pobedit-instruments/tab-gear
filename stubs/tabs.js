export default [
	{
		title : 'Google Cloud Computing, Hosting Services & APIs | Google Cloud Platform',
		url   : 'https://cloud.google.com',
		favIconUrl: 'https://cloud.google.com/_static/cc7442d8c7' +
					'/images/cloud/icons/favicons/favicon.ico',
		id : 0,
		index : 0,
		active: true,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Plus +++++++++++++++++++++++++++++++++++++++++++',
		url   : 'https://developers.google.com/+/',
		favIconUrl: 'https://developers.google.com/site-assets/logo-google-plus-white.svg',
		id : 1,
		index : 1,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Play',
		url   : 'https://play.google.com',
		favIconUrl: 'https://developers.google.com/_static/cc7442d8c7/images/favicon.png',
		id : 2,
		index : 2,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Maps',
		url   : 'https://maps.google.com',
		favIconUrl: 'https://www.google.com/images/branding/product/ico/maps_32dp.ico',
		id : 3,
		index : 3,
		active: false,
		highlighted: false,
		incognito: true
	},

	{
		title : 'Google Translate',
		url   : 'https://translate.google.ru/',
		favIconUrl: 'https://translate.google.ru/favicon.ico',
		id : 4,
		index : 4,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Android',
		url   : 'https://developer.android.com',
		favIconUrl: 'https://developer.android.com/favicon.ico',
		id : 5,
		index : 5,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Chrome',
		url   : 'https://google.com/chrome',
		favIconUrl: 'https://www.google.com/images/icons/product/chrome-32.png',
		id : 6,
		index : 6,
		highlighted: true,
		active: false
	},

	{
		title : 'Google Developers',
		url   : 'https://developers.google.com/web/',
		favIconUrl: 'https://developers.google.com/web/images/web-fundamentals-icon192x192.png',
		id : 7,
		index : 7,
		active: false,
		highlighted: false,
		incognito: false
	},

	{
		title : 'Google Firebase',
		url   : 'https://firebase.google.com/',
		favIconUrl: 'https://firebase.google.com/_static/cc7442d8c7/images/firebase/favicon.png',
		id : 9,
		index : 9,
		highlighted: false,
		active: false
	}
];
