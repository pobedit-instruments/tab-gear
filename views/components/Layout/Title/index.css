@import "../../decor.css";

:root {
	--size: 26px;
}

.tg-title {
	background-color: var(--tab-title-background-color);
	box-shadow:
			0 2px 2px 0 rgba(0, 0, 0, .14),
			0 3px 1px -2px rgba(0, 0, 0, .2),
			0 1px 5px 0 rgba(0, 0, 0, .12);
	color: var(--tab-title-color);
	cursor: default;
	border-radius: 1px;
	font-size: 20px;
	margin: 25px auto 0 auto;
	padding: var(--padding) 43px var(--padding) 55px;
	position: relative;
	user-select: none;
}

.tg-title,
.tg-list {
	border: var(--tab-title-border-color);
}

.tg-title__logo {
	background: url('../../../../files/icons/38x38.png') 0 0 no-repeat;
	background-size: var(--size);
	margin-right: 13px;
	top: 12px;
	left: var(--padding);
	position: absolute;
	width: var(--size);
	height: var(--size);
}

.tg-controls__more {
	border-radius: 3px;
	cursor: pointer;
	color: var(--tab-title-color);
	overflow: hidden;
	outline: none;
	position: absolute;
	top: 14px;
	right: 7px;
	width: 22px;
	text-align: center;
	transition: background-color .3s ease-in-out;

	&:after {
		content: 'more_vert';
	}

	&:hover {
		background-color: #1490f1;
	}

	&:active {
		box-shadow: inset 0 0 11px rgba(53, 134, 202, 0.6);
	}
}
