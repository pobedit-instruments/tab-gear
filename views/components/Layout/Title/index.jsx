import React, { Component, PropTypes } from 'react';
import BEMHelper from 'react-bem-helper';
import { bind } from 'decko';

import * as ActionTypes from '../../../constants/ActionTypes';
import './index.css';

class Title extends Component {
	constructor (properties) {
		super(...arguments);

		this.class = new BEMHelper({
			name: 'tg-title'
		});

		this.state = {
			credentials: false
		};
	}

	/**
	 * Show credentials
	 *
	 * @param {Event} event
	 */
	@bind
	showCredentials (event) {
		let { actions, type } = this.props;

		let types = [
			ActionTypes.SHOW_TABS,
			ActionTypes.SEARCH_TABS,
			ActionTypes.DISABLE_TABS
		];

		let active = types.includes(type);

		if (active) {
			actions.Layout.showCredentials();
		}
		else {
			actions.Tabs.showTabs();
		}

		this.setState({ credentials: active });

		event.preventDefault();
	}

	getTitle (type, tabs) {
		let title = [];

		switch (type) {
			case ActionTypes.SHOW_CREDENTIALS:
			case ActionTypes.SHOW_ERRORS:
			case ActionTypes.DISCARD_TABS:
			case ActionTypes.TABS_LOCKED:
			case ActionTypes.SEARCH_TABS:
				let { length } = tabs.search;

				title = [type, length];

				if (type !== ActionTypes.SEARCH_TABS || length) {
					break;
				}

			default:
				title = [ActionTypes.SHOW_TABS, tabs.actual.length];
		}

		let [ key, value ] = title;

		return chrome.i18n.getMessage(key.toLowerCase(), [ value ]);
	}

	render () {
		let { tabs, type } = this.props;

		return <div { ...this.class() }>
					<div { ...this.class('logo') } />

					<div { ...this.class('text') }>
						{ this.getTitle(type, tabs) }
					</div>

					<a href="#" tabIndex="1" className="tg-icon tg-controls__more"
					   onClick={ this.showCredentials } />
				</div>;
	}
}

Title.propTypes = {
	tabs: PropTypes.object.isRequired,
	type: PropTypes.string.isRequired
};

export default Title;
