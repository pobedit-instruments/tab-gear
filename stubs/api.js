import chrome from 'chrome-stub';

import tabs from './tabs';
import locale from '../locale/en/messages.json';

chrome.i18n._locales = locale;

let methods = [
	'query',
	'remove',
	'move',
	'create',
	'discard',
	'highlight'
];

for (let method of methods) {
	chrome.tabs[method].yields(tabs);
}

export default chrome;
