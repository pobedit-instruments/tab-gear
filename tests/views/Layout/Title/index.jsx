import React from 'react';
import renderer from 'react-test-renderer';
// import { shallow, mount, render } from 'enzyme';

import Title from '../../../../views/components/Layout/Title';
import * as props from '../../../../stubs/props';

test('Layout::Title view', () => {
	let component = renderer.create(<Title { ...props } />);
	let view = component.toJSON();

	let actual = expect(view);

	actual.toMatchSnapshot();
});
