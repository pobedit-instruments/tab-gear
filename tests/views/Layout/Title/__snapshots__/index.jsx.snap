exports[`test Layout::Title view 1`] = `
<div
  className="tg-title">
  <div
    className="tg-title__logo" />
  <div
    className="tg-title__text">
    Active tabs: 9
  </div>
  <a
    className="tg-icon tg-controls__more"
    href="#"
    onClick={[Function]}
    tabIndex="1" />
</div>
`;
